src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -556,5 +556,162 @@ public AggregateContext(T seed)
+                    dt.Rows.Add(new object[] { "System.Decimal","decimal",7 });
+                    dt.Rows.Add(new object[] { "System.Decimal","numeric",7 });
+                    dt.Rows.Add(new object[] { "System.Boolean","bit",3 });
+                    dt.Rows.Add(new object[] { "System.Boolean","yesno",3 });
+                    dt.Rows.Add(new object[] { "System.Boolean","logical",3 });
+                    dt.Rows.Add(new object[] { "System.Boolean","bool",3 });
+                    dt.Rows.Add(new object[] { "System.Boolean","boolean",3 });
+                    dt.Rows.Add(new object[] { "System.Byte","tinyint",2 });
+                    dt.Rows.Add(new object[] { "System.Int64","integer",12 });
+                    dt.Rows.Add(new object[] { "System.Int64","counter",12 });
+                    dt.Rows.Add(new object[] { "System.Int64","autoincrement",12 });
+                    dt.Rows.Add(new object[] { "System.Int64","identity",12 });
+                    dt.Rows.Add(new object[] { "System.Int64","long",12 });
+                    dt.Rows.Add(new object[] { "System.Int64","bigint",12 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","binary",1 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","varbinary",1 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","blob",1 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","image",1 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","general",1 });
+                    dt.Rows.Add(new object[] { "System.Byte[]","oleobject",1 });
+                    dt.Rows.Add(new object[] { "System.String","varchar",16 });
+                    dt.Rows.Add(new object[] { "System.String","nvarchar",16 });
+                    dt.Rows.Add(new object[] { "System.String","memo",16 });
+                    dt.Rows.Add(new object[] { "System.String","longtext",16 });
+                    dt.Rows.Add(new object[] { "System.String","note",16 });
+                    dt.Rows.Add(new object[] { "System.String","text",16 });
+                    dt.Rows.Add(new object[] { "System.String","ntext",16 });
+                    dt.Rows.Add(new object[] { "System.String","string",16 });
+                    dt.Rows.Add(new object[] { "System.String","char",16 });
+                    dt.Rows.Add(new object[] { "System.String","nchar",16 });
+                    dt.Rows.Add(new object[] { "System.DateTime","datetime",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","smalldate",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","timestamp",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","date",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","time",6 });
+                    dt.Rows.Add(new object[] { "System.Guid","uniqueidentifier",4 });
+                    dt.Rows.Add(new object[] { "System.Guid","guid",4 });
```

Comment:
We need to refine this information before merging.

Reply:
The information here describes what types of .NET can be mapped to types that will be understood by the database driver, and the corresponding System.Data.Common DbTypes. This information is taken from the System.Data.SQLite.dll library.

I think this is more design decision. Do you want to continue from here?
I'm already happy with the current implementation but feel free to change what ever you need.